pkg/cli/serve/serve.go

```diff
@@ -141,7 +142,9 @@ func NewServeCmd() *cobra.Command {
 func (o *serveOptions) Complete(cmd *cobra.Command, args []string) error {
 
 	// ReadFlagsFromFile would read the values of flags from files (if any)
-	err := ReadFlagsFromFile(cmd, config.AccessToken, config.Services, config.OcmURL, config.ClusterID)
+	// Don't include 'services' in flags from file defintion, as it is a string array.
+	err := ReadFlagsFromFile(cmd, config.AccessToken, config.OcmURL, config.ExternalClusterID)
```

Comment:
Currently, we pass the supported services via `configmap/ocm-agent-cm`. 
What it would be if we remove the config.Services here?

Or can we implement it as the same? We can update the configmap.data.services as array?

Reply:
@bmeng I've updated `ReadFlagsFromFile` to support loading services as a file. 
You can test it like so: 
Change `test/build-and-run.sh` like so:
```
${GIT_ROOT}/build/_output/ocm-agent serve --access-token "$ACCESS_TOKEN" --services "@service_logs_file" --cluster-id "$EXT_CLUSTERID" --ocm-url ${OCM_BASE_URL}
```
Create 'service_logs_file' in `ocm-agent` root:
```
[17:09:28] hectorkemp:ocm-agent git:(make-services-flag-string-array*) $ cat <<EOF > service_logs_file
service_logs
EOF
```
Run `test/build-and-run.sh cluster-name`. It will load it from your `service_logs_file`.
```
[17:11:37] hectorkemp:ocm-agent git:(make-services-flag-string-array*) $ test/build-and-run.sh cluster

--- Fetching Cluster ID and creating temporary KUBECONFIG...

--- Building ocm-agent locally...
make: Entering directory '/home/hectorkemp/redhat/go/ocm-agent'
?       github.com/openshift/ocm-agent/cmd/ocm-agent    [no test files]
?       github.com/openshift/ocm-agent/pkg/cli  [no test files]
?       github.com/openshift/ocm-agent/pkg/cli/serve    [no test files]
?       github.com/openshift/ocm-agent/pkg/config       [no test files]
?       github.com/openshift/ocm-agent/pkg/consts       [no test files]
?       github.com/openshift/ocm-agent/pkg/consts/test  [no test files]
?       github.com/openshift/ocm-agent/pkg/handlers/mocks       [no test files]
?       github.com/openshift/ocm-agent/pkg/k8s  [no test files]
?       github.com/openshift/ocm-agent/pkg/logging      [no test files]
ok      github.com/openshift/ocm-agent/pkg/handlers     (cached)
?       github.com/openshift/ocm-agent/pkg/util/test/generated/mocks/client     [no test files]
?       github.com/openshift/ocm-agent/pkg/ocm  [no test files]
?       github.com/openshift/ocm-agent/pkg/util/test/mockgenerator/client       [no test files]
ok      github.com/openshift/ocm-agent/pkg/metrics      (cached)
mkdir -p build/_output
GOOS=linux GOARCH=amd64 CGO_ENABLED=1 GOEXPERIMENT=boringcrypto GOFLAGS= go build -gcflags="all=-trimpath=/home/hectorkemp/redhat/go" -asmflags="all=-trimpath=/home/hectorkemp/redhat/go" -tags="fips_enabled" -o build/_output/ocm-agent ./cmd/ocm-agent
make: Leaving directory '/home/hectorkemp/redhat/go/ocm-agent'

--- Running ocm-agent locally...
(Keep this terminal open to follow log of ocm-agent. Open new terminal and run test scripts.)
Link: http://localhost:8081

/home/hectorkemp/redhat/go/ocm-agent
***** Starting with FIPS crypto enabled *****
INFO[2023-11-01T17:11:58+11:00] Starting ocm-agent server                    
INFO[2023-11-01T17:11:58+11:00] Fleet mode not configured                     FleetMode=false
INFO[2023-11-01T17:11:58+11:00] Start listening on metrics port               Port=8383
INFO[2023-11-01T17:11:58+11:00] Connection with OCM initialised successfully in non-fleet mode 
INFO[2023-11-01T17:11:58+11:00] Initialising alertmanager webhook handler in NON-fleet mode 
INFO[2023-11-01T17:11:58+11:00] Start listening on service port               Port=8081
```

Reply:
Either way, it works before/after with `build-and-run`, which reads from OCM agent config.